src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/PermissionCheckService.java

```diff
@@ -0,0 +1,83 @@
+package uk.gov.hmcts.reform.wataskmanagementapi.services;
+
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.wataskmanagementapi.auth.permission.PermissionEvaluatorService;
+import uk.gov.hmcts.reform.wataskmanagementapi.auth.permission.entities.PermissionTypes;
+import uk.gov.hmcts.reform.wataskmanagementapi.auth.role.RoleAssignmentService;
+import uk.gov.hmcts.reform.wataskmanagementapi.auth.role.entities.Assignment;
+import uk.gov.hmcts.reform.wataskmanagementapi.domain.entities.camunda.CamundaVariable;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import static uk.gov.hmcts.reform.wataskmanagementapi.auth.permission.entities.PermissionTypes.EXECUTE;
+import static uk.gov.hmcts.reform.wataskmanagementapi.auth.permission.entities.PermissionTypes.MANAGE;
+import static uk.gov.hmcts.reform.wataskmanagementapi.auth.permission.entities.PermissionTypes.OWN;
+
+@Component
+public class PermissionCheckService {
+
+    private final IdamService idamService;
+    private final PermissionEvaluatorService permissionEvaluatorService;
+    private final RoleAssignmentService roleAssignmentService;
+    private final CamundaService camundaService;
+
+    public PermissionCheckService(IdamService idamService,
```

Comment:
@Autowired missing?

Reply:
This one is optional, Vam. As per spring boot doc: 
 _If a class only declares a single constructor, to begin with, it will always be used, even if not annotated. An annotated constructor does not have to be public._